Tag Database Model Change

smithWEBtek opened the pull request:
* Add Tag and Tagging refinery extensions
* Add relationships to Announcement and Event models
* Style tags and taggings admin interfaces
* Add functionality to creaet taggings for events and announcements
* Create tag and tagging model tests

Resolves #332 Tag Database Model Change

Why is this change necessary?
This is required to provide tagging functionality on a variety of database objects.

How does it address the issue?
The Tag and Taggings models allow tags to be assigned and retrieved for Announcement and Event models. These database additions will feed the functionality required to filter the "Find Out" page using the taggings records.

What side effects does it have?
None.

mzagaja commented:
Can you merge #335 and then rebase this? I'm getting lots of merge conflicts when I try doing it locally and currently am experiencing two test failures, one of which I thought was fixed in #335 so want to confirm. Here are my test failures:

```
Failures:

  1) Weigh In Stories sizes flexbox container correctly with 4 objects
     Failure/Error: expect(page).to have_xpath('//section[@id="stories"][@style="height: 837px;"]')
       expected to find xpath "//section[@id=\"stories\"][@style=\"height: 837px;\"]" but there were no matches

     [Screenshot]: tmp/screenshots/failures_r_spec_example_groups_weigh_in_stories_sizes_flexbox_container_correctly_with_4_objects_505.png


     # ./spec/system/weigh_in_stories_spec.rb:10:in `block (2 levels) in <main>'

  2) One boxes does not display hidden boxes
     Failure/Error: expect(all('.styled-box__title')[0]&.text).not_to have_text(one_box1.title)
       expected not to find text "" in ""
```

smithWEBtek commented:
> Can you merge #335 and then rebase this? I'm getting lots of merge conflicts when I try doing it locally and currently am experiencing two test failures, one of which I thought was fixed in #335 so want to confirm. Here are my test failures:
> 
> ```
> Failures:
> 
>   1) Weigh In Stories sizes flexbox container correctly with 4 objects
>      Failure/Error: expect(page).to have_xpath('//section[@id="stories"][@style="height: 837px;"]')
>        expected to find xpath "//section[@id=\"stories\"][@style=\"height: 837px;\"]" but there were no matches
> 
>      [Screenshot]: tmp/screenshots/failures_r_spec_example_groups_weigh_in_stories_sizes_flexbox_container_correctly_with_4_objects_505.png
> 
> 
>      # ./spec/system/weigh_in_stories_spec.rb:10:in `block (2 levels) in <main>'
> 
>   2) One boxes does not display hidden boxes
>      Failure/Error: expect(all('.styled-box__title')[0]&.text).not_to have_text(one_box1.title)
>        expected not to find text "" in ""
> ```




I have rebased this (tags) PR on develop. 
And the conflicting test fixes are now removed ( to be addressed separately ). 
Those 2 tests seem to be failing / passing inconsistently, with the possible exception of the height: 837px change solving the 1st test. 

So this PR now has Reports (without tagging functionality yet) and the Tags and Taggings code.
